File: Catalog/Catalog.Host/Data/Entities/Laptop.cs
namespace Catalog.Host.Data.Entities
{
    public class Laptop : IBaseEntity
    {
        public int Id { get; set; }

        public string Name { get; set; } = null!;

        public string Description { get; set; } = null!;

        public decimal Price { get; set; }

        public string PictureFileName { get; set; } = null!;

        public int LaptopBrandId { get; set; }

        public LaptopBrand LaptopBrand { get; set; } = null!;

        public int ScreenTypeId { get; set; }

        public LaptopScreenType ScreenType { get; set; } = null!;

        public int AvailableStock { get; set; }
    }
}

File: Web/SPA/src/laptop/models/laptop.ts
import { Brand } from "./brand";
import { ScreenType } from "./screenType";

export interface Laptop {
    id: number;
    name: string;
    description: string;
    price: number;
    pictureUrl: string;
    laptopBrand: Brand;
    screenType: ScreenType;
    availableStock: number;
    sku: string;
}

export type LaptopFilter = {
    laptopBrandId?: number;
    screenTypeId?: number;
}